zdx/mem.go

```diff
@@ -1,59 +1,109 @@
 package zdx
 
 import (
-	"errors"
+	"fmt"
 	"sort"
+
+	"github.com/brimsec/zq/zcode"
+	"github.com/brimsec/zq/zng"
+	"github.com/brimsec/zq/zng/resolver"
 )
 
-// MemTable implements a simple stub for an in-memory SST Reader.  This is
-// useful for testing and reference.  An SST client will typically implement its
-// own in-memory data structure that provides a Reader implementation to convert
-// custom key/value pairs into string/byte-slice pairs.
+// MemTable implements an in-memory table to build a zdx.
+// It implements zbuf.Reader and will generate a stream of zng.Records that
+// are either single column ("key") or a two-column ("key", "value") where the
+// types of the columns depend upon the zng.Values entered into the table.
 type MemTable struct {
-	table  map[string][]byte
-	keys   []string
-	offset int
+	table   map[string]zcode.Bytes
+	keys    []string
+	offset  int
+	zctx    *resolver.Context
+	builder *zng.Builder
+	recBuf  zng.Record
```

Comment:
looks like `recBuf` isn't used ?

Reply:
yeah, that got moved into zng.Builder.  Will delete.